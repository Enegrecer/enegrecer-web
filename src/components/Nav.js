import React from 'react';
import { withHistoryPropTypes } from './sharedPropTypes';

const Nav = () => (
    <ul className='right hide-on-med-and-down'>
      <li><a name="sobre" href="/sobre">Sobre</a></li>
    </ul>
);

Nav.propTypes = withHistoryPropTypes();

export default Nav;
