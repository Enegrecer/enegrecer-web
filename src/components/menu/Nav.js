import React from 'react';

const Nav = () => (
    <ul className='right hide-on-med-and-down'>
      <li><a name="sobre" href="/sobre">Sobre</a></li>
    </ul>
);

export default Nav;
