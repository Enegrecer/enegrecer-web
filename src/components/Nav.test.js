import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Nav from './Nav';

describe('Nav', () => {
  const historyMock = { push: jest.fn() };
  const wrapper = shallow(
    <Nav history={historyMock} />,
  );

  it('possui a mesma estrutura', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('tem a classe right hide-on-med-and-down', () => {
    expect(wrapper.hasClass('right hide-on-med-and-down')).toBeTruthy();
  });

  it('tem o link de sobre', () => {
    expect(wrapper.find('[href="/sobre"]').exists()).toBeTruthy();
  });
});
