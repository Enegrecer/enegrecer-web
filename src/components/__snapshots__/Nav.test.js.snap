// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav possui a mesma estrutura 1`] = `
<ul
  className="right hide-on-med-and-down"
>
  <li>
    <a
      href="/sobre"
      name="sobre"
    >
      Sobre
    </a>
  </li>
</ul>
`;
